bundles/org.openhab.core.io.rest.transform/src/main/java/org/openhab/core/io/rest/transform/internal/TransformationConfigurationResource.java

```diff
@@ -0,0 +1,192 @@
+        if (configuration == null) {
+            return Response.status(Response.Status.NOT_FOUND).build();
+        }
+        return Response.ok(configuration).build();
+    }
+
+    @PUT
+    @Path("configuration/{uid: .*}")
+    @Consumes(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
+    @Operation(operationId = "putTransformationConfiguration", summary = "Get a single transformation configuration", responses = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "405", description = "Configuration not editable"),
+            @ApiResponse(responseCode = "400", description = "Bad Request (content missing or invalid)") })
+    public Response putTransformationConfiguration(
+            @PathParam("uid") @Parameter(description = "Configuration UID") String uid,
+            @Parameter(description = "configuration", required = true) @Nullable TransformationConfigurationDTO newConfiguration) {
+        logger.debug("Received HTTP PUT request at '{}'", uriInfo.getPath());
+
+        TransformationConfiguration oldConfiguration = transformationConfigurationRegistry.get(uid);
+        if (oldConfiguration != null && !isEditable(uid)) {
+            return Response.status(Response.Status.METHOD_NOT_ALLOWED).build();
+        }
+
+        if (newConfiguration == null) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("Content missing.").build();
+        }
+
+        if (!uid.equals(newConfiguration.uid)) {
+            return Response.status(Response.Status.BAD_REQUEST).entity("UID of configuration and path not matching.")
+                    .build();
+        }
+
+        TransformationConfiguration transformationConfiguration = new TransformationConfiguration(newConfiguration.uid,
+                newConfiguration.label, newConfiguration.type, newConfiguration.language, newConfiguration.content);
+        try {
+            if (oldConfiguration != null) {
+                managedTransformationConfigurationProvider.update(transformationConfiguration);
+            } else {
+                managedTransformationConfigurationProvider.add(transformationConfiguration);
+            }
+        } catch (IllegalArgumentException e) {
+            return Response.status(Response.Status.BAD_REQUEST).entity(Objects.requireNonNullElse(e.getMessage(), ""))
+                    .build();
+        }
+
+        return Response.ok().build();
+    }
+
+    @DELETE
+    @Path("configuration/{uid: .*}")
+    @Produces(MediaType.TEXT_PLAIN)
+    @Operation(operationId = "deleteTransformationConfiguration", summary = "Get a single transformation configuration", responses = {
+            @ApiResponse(responseCode = "200", description = "OK"),
+            @ApiResponse(responseCode = "405", description = "Configuration not editable"),
+            @ApiResponse(responseCode = "404", description = "UID not fond") })
```

Comment:
"UID not fond" has a typo and it would also be nice when the `@ApiReponse`s are sorted on the `responseCode` in all the new operations.

```suggestion
            @ApiResponse(responseCode = "200", description = "OK"),
            @ApiResponse(responseCode = "404", description = "UID not found"),
            @ApiResponse(responseCode = "405", description = "Configuration not editable") })
```